### PR TITLE
[Order creation] Display prices only for synced items

### LINKED ISSUE #10521: [Order creation] Display prices only for synced items
In case the prices of products are set up to be entered inclusive of taxes there is a glitch in the order creation. When adding an item to the order the price (inclusive of taxes) gets overridden once the item gets synced. We can enhance the UX by displaying prices only for the already synced items. 

| prices entered including taxes | prices entered excluding taxes |
|---|---|
| <video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/ca0499ec-8394-4048-940a-cfaf63f8543a width=300/> | <video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/43fed1f5-4a2e-46a2-873d-6f4a30bfb4b8 width=300/> |

### PR DESCRIPTION
**Pull request by backwardstruck**
Closes: #10521

### Description

This pull request addresses the issue #10521 opened by @samiuelson. The enhancement focuses on improving the UX during order creation when prices of products are inclusive of taxes. The goal is to display prices only for items that are already synced, eliminating a glitch where the price gets overridden during the syncing process.

### Testing instructions
1. Verify that prices are displayed accurately for items that are already synced.
2. Confirm that the final price (e.g., CA$32.27) is shown without changes during the syncing process.
3. Test scenarios where prices are entered including and excluding taxes.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by backwardstruck** [comment]
@kidinov what do you think about how I'm hiding the non-synced price? I just changed the color but maybe the text view shouldn't be created at all until it's ready.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 1]
```diff
@@ -214,7 +214,11 @@ fun ExpandableProductCard(
                     ),
                 style = MaterialTheme.typography.body2,
                 text = getQuantityWithTotalText(product),
-                color = colorResource(id = R.color.color_on_surface_disabled)
+                color = if (product.item.isSynced()) {
+                    colorResource(id = R.color.color_on_surface_disabled)
+                } else {
+                    colorResource(id = com.google.android.material.R.color.mtrl_btn_transparent_bg_color)
+                }
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Maybe there's a better color I should use here or instead I could not not create the view in the first place.

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: 👋

I'd go with not rendering the view in the first place, as this, in my opinion, is more "compose" idiomatic way of doing things.  

```
if (product.item.isSynced()) {
  Text() {
   }
   Text(){
   }
```
If the card will be "jumping", consider using `Spacer` component

Also, regarding color - we usually try to use the colors from our own palette, not the ones from the material library

Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Good suggestions! Updated.

**Comment by backwardstruck** [comment]
Thanks @kidinov This should not be ready for review again (changed from `draft` to `open`.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by kidinov — APPROVED** [review]
LGTM, thanks!

Only what I would consider adding is this case as a new compose @Preview function, to make it easier to debug when needed

**Comment by backwardstruck** [comment]
> LGTM, thanks!
> 
> Only what I would consider adding is this case as a new compose @Preview function, to make it easier to debug when needed

Since the PR was auto merged I created a separate issue which I will work on here:
https://github.com/woocommerce/woocommerce-android/issues/10712

**Comment by backwardstruck** [comment]
Updated milestone to 17.3 since it was merged this week.

**Pull request merged**